web/src/beta/features/PluginPlayground/Code/hook.ts

```
@@ -0,0 +1,131 @@
+        const { zone, section, area } = cur.widgetLayout?.defaultLocation ?? {
+          zone: "outer",
+          section: "left",
+          area: "top"
+        };
+
+        const zoneAlignSystem = prv?.alignSystem?.[zone] ?? {};
+        const sectionAlignSystem = zoneAlignSystem[section] ?? {};
+        const areaAlignSystem = sectionAlignSystem[area] ?? { widgets: [] };
+
+        return {
+          ...prv,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)